src/tzdb.rs

```diff
@@ -0,0 +1,457 @@
+//  - SystemTimeZoneIdentifier
+//  - IsTimeZoneOffsetString
+//  - GetNamedTimeZoneEpochNanoseconds
+//     - fn(id, isoDateTimeRecord) -> [epochNanoseconds]
+//  - GetNamedTimeZoneOffsetNanoseconds
+//     - fn(id, epochNanoseconds) -> [offset]
+
+// TODO: Potentially implement a IsoDateTimeRecord type to decouple
+// public facing APIs from IsoDateTime
+
+// Could return type be something like [Option<i128>; 2]
+
+// NOTE: tzif data is computed in glibc's `__tzfile_compute` in `tzfile.c`.
+//
+// Handling the logic here may be incredibly important for full tzif support.
+
+// NOTES:
+//
+// Transitions to DST (in march) + 1. Empty list between 2:00-3:00.
+// Transitions to Std (in nov) -1. Two elements 1:00-2:00 is repeated twice.
+
+// Transition Seconds + (offset diff)
+// where
+// offset diff = is_dst { dst_off - std_off } else { std_off - dst_off }, i.e. to_offset - from_offset
+
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+
+use tzif::{
+    self,
+    data::{
+        time::Seconds,
+        tzif::{DataBlock, LocalTimeTypeRecord, TzifData},
+    },
+};
+
+use crate::{components::tz::TzProvider, iso::IsoDateTime, TemporalError, TemporalResult};
+
+const ZONEINFO_DIR: &str = "/usr/share/zoneinfo/";
+
+pub type TransitionInfo = [Option<LocalTimeTypeRecord>; 2];
+
+#[derive(Debug)]
+pub struct Tzif(TzifData);
+
+impl Tzif {
+    fn read_tzif(identifier: &str) -> TemporalResult<Self> {
+        let mut path = PathBuf::from(ZONEINFO_DIR);
+        path.push(identifier);
+        Self::from_path(&path)
+    }
+
+    pub fn from_path<P: AsRef<Path>>(path: P) -> TemporalResult<Self> {
+        tzif::parse_tzif_file(path)
+            .map(Self)
+            .map_err(|e| TemporalError::general(e.to_string()))
+    }
+
+    // There are ultimately
+    pub fn get(&self, epoch_seconds: &Seconds) -> TemporalResult<LocalTimeTypeRecord> {
+        let Some(result) = self.binary_search(epoch_seconds) else {
+            return Err(TemporalError::general("Only Tzif v2+ is supported."));
+        };
+
+        let db = self
+            .0
+            .data_block2
+            .as_ref()
+            .expect("binary search throws error if datablock doesn't exist.");
+
+        match result {
+            Ok(idx) => Ok(get_local_record(db, idx - 1)),
+            Err(idx) if idx == 0 => Ok(get_local_record(db, idx)),
+            Err(idx) => {
+                if db.transition_times.len() <= idx {
+                    return Err(TemporalError::general("TODO: Support POSIX tz string."));
+                }
+                Ok(get_local_record(db, idx - 1))
+            }
+        }
+    }
+
+    // There are various other ways to search rather than binary_search. See glibc
+    pub fn binary_search(&self, epoch_seconds: &Seconds) -> Option<Result<usize, usize>> {
+        self.0
+            .data_block2
+            .as_ref()
+            .map(|b| b.transition_times.binary_search(epoch_seconds))
+    }
```

Comment:
Yeah, I can add that method. I think my primary thoughts were that it would be nice to try and fix this in the `tzif` crate. But in the meantime, adding a method makes sense.